iset.mm

```diff
@@ -147954,6 +147981,516 @@ of the multiplicative monoid ( ~ df-mgp ) of a ring-like structure.  This
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Semirings
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c SRing $.
+
+  $( Extend class notation with the class of all semirings. $)
+  csrg $a class SRing $.
+
+  ${
+    $d f n p r t x y z $.
+    $( Define class of all semirings.  A semiring is a set equipped with two
+       everywhere-defined internal operations, whose first one is an additive
+       commutative monoid structure and the second one is a multiplicative
+       monoid structure, and where multiplication is (left- and right-)
+       distributive over addition.  Compared to the definition of a ring, this
```

Comment:
"Compared to the definition of a ring, this definition also adds that the additive identity is an absorbing element of the multiplicative law," This sounds as if a semiring is a specialization of a ring, which is not true. @tirix do you have an idea, how we can find a clearer wording?

Reply:
Maybe "_like with rings, the additive identity is an absorbing element of the multiplicative law, but in the case of semirings, this has to be part of the definition, as it cannot be deduced from distributivity alone._"

Reply:
Thanks for the suggestion, I'll use that wording. I really didn't understand what the original was trying to say but now that you suggest this, it does seem like that's what it was attempting to say.